frontend/src/metabase/core/components/Radio/Radio.tsx

```diff
@@ -45,7 +46,7 @@ export interface RadioProps<TValue, TOption = RadioOption<TValue>>
   value?: TValue;
   options: TOption[];
   optionKeyFn?: (option: TOption) => Key;
-  optionNameFn?: (option: TOption) => string;
+  optionNameFn?: (option: TOption) => ReactNode;
```

Comment:
I'm honestly not sure when to use `JSX.Element` vs `ReactNode` vs `ReactElement` and I somewhat have seen these 3 being used interchangeably. What's your take on this?